core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -1886,6 +1924,10 @@ public NodeRepresentation addNode(
             List<PlanCostEstimate> estimatedCosts = allNodes.stream()
                     .map(nodeId -> estimatedStatsAndCosts.getCosts().getOrDefault(nodeId, PlanCostEstimate.unknown()))
                     .collect(toList());
+            String argumentName = argumentNames.get(rootNode.getId());
```

Comment:
Instead of passing the whole map of node id -> name and having this method look it up, it would be cleaner to pass the intended name directly. At the end of the day, for each branch of recursion when visiting a child, there's only one name that's valid.

This means that the Context just just contain a single name that represents the "tag" associated with the subtree. If present, it will be prepended to the root of that subtree when printed. It also makes it more generic and allows for tagging any subtree (e.g., for a JOIN, we might want to tag the left and right differently), etc.

Reply:
If the context had just a single tag instead of the map, I would have to process sources in a loop, creating a new context per child instead of just call `processChildren`.

However, my main reason for using a map by id was to make it more robust. The usual usage pattern for context is that we process the node, and then recursively process children, passing the same context. In the current design I can safely do it, as node ids are unique. If the context had just a tag, we would need to explicitly clear the context before the recursive call. It would be unintuitive for future contributors.